Use `unified-engine` (#13)
unifiedjs/unified-language-server: README.md

> # unified-language-server

Author: I copied the readme from `rehype-format` and adjusted it to match the new readme format used throughout unified.

# unified-language-server

[![Build][build-badge]][build]
[![Coverage][coverage-badge]][coverage]
[![Downloads][downloads-badge]][downloads]
[![Size][size-badge]][size]
[![Sponsors][sponsors-badge]][collective]
[![Backers][backers-badge]][collective]
[![Chat][chat-badge]][chat]

Create a **[language server][]** based on **[unified][]** ecosystems.

## Contents

*   [What is this?](#what-is-this)
*   [When should I use this?](#when-should-i-use-this)
*   [Install](#install)
*   [Use](#use)
*   [API](#api)
    *   [`createUnifiedLanguageServer(options)`](#createunifiedlanguageserveroptions)
*   [Examples](#examples)
*   [Types](#types)
*   [Compatibility](#compatibility)
*   [Related](#related)
*   [Contribute](#contribute)
*   [License](#license)

## What is this?

This package exports a function which can be used to create a
[language server][] based on [unified][] processors.
It can do the following:

*   format documents based on a unified processor
*   validate documents based on a unified processor

> *   support configuration files (such as `.remarkrc`) using [unified-engine][]

Reviewer: question on scope, is there interest in having this also be able to generate previews of rendered documents inside vscode? (for example `remark-parse`, with `remark-rehype`, and `rehype-stringify`)

Author: Possibly. That’s not just a simple editor feature though. I think this would require a way to implement custom command, then custom logic in the client (`vscode-remark`) to implement this.

So this might be interesting for later.

# unified-language-server

[![Build][build-badge]][build]
[![Coverage][coverage-badge]][coverage]
[![Downloads][downloads-badge]][downloads]
[![Size][size-badge]][size]
[![Sponsors][sponsors-badge]][collective]
[![Backers][backers-badge]][collective]
[![Chat][chat-badge]][chat]

Create a **[language server][]** based on **[unified][]** ecosystems.

## Contents

*   [What is this?](#what-is-this)
*   [When should I use this?](#when-should-i-use-this)
*   [Install](#install)
*   [Use](#use)
*   [API](#api)
    *   [`createUnifiedLanguageServer(options)`](#createunifiedlanguageserveroptions)
*   [Examples](#examples)
*   [Types](#types)
*   [Compatibility](#compatibility)
*   [Related](#related)
*   [Contribute](#contribute)
*   [License](#license)

## What is this?

This package exports a function which can be used to create a
[language server][] based on [unified][] processors.
It can do the following:

*   format documents based on a unified processor
*   validate documents based on a unified processor

> *   support configuration files using [unified-engine][]

Reviewer 2: nit: very few people know unified engine because it's a low-level dependency, but they do know the config files. 

```suggestion
*   support configuration files (such as `.remarkrc`) using [unified-engine][]
```

**unified** is a project that validates and transforms content with abstract
syntax trees (ASTs).

> **unified-engine** is a project that can apply configure ecosystems based on
> unified on a workspace using configuration files.

Reviewer 2: ```suggestion
**unified-engine** is a project to configure unified ecosystems based on
unified on a workspace using configuration files.
```

###### `options.defaultSource`

Default source used for diagnostics (`string`, optional)

###### `options.ignoreName`

Name of ignore files to load (`string`, optional)

###### `options.packageField`

Property at which configuration can be found in package.json files (`string`,
optional)

###### `options.pluginPrefix`

Optional prefix to use when searching for plugins (`string`, optional)

###### `options.plugins`

Plugins to use by default (`Array|Object`, optional)

Typically this contains 2 plugins named `*-parse` and `*-stringify`.

###### `options.rcName`

Name of configuration files to load (`string`, optional)

## Examples

For examples, see the following projects:

*   [redot-language-server][]
*   [rehype-language-server][]
*   [remark-language-server][]

## Types

This package is fully typed with [TypeScript][].
It exports an `Options` type, which specifies the interface of the accepted
options.

## Compatibility

Projects maintained by the unified collective are compatible with all maintained
versions of Node.js.
As of now, that is Node.js 12.20+, 14.14+, and 16.0+.
Our projects sometimes work with older versions, but this is not guaranteed.

> This plugin works with `unified` version 10+.

Author: Not sure about this. The language server brings its own unified dependency. It relies on the user to install the appropriate plugins.

Reviewer 2: I suppose the line can be removed then

Reviewer 3: my thought was that the tool comes with core functionality. Any add-ons are user supplied.

Author: > my thought was that the tool comes with core functionality. Any add-ons are user supplied.

This is exactly what `unified-engine` does. :)

## Related

*   [unified][]
    — create pipeline for working with syntax trees
*   [unified-args][]
    — create a CLI for a unified pipeline

## Contribute

See [`contributing.md`][contributing] in [`unified/.github`][health] for ways
to get started.
See [`support.md`][support] for ways to get help.

This project has a [code of conduct][coc].
By interacting with this repository, organization, or community you agree to
abide by its terms.

## License

> [MIT][license] © [Titus Wormer][author]

Author: Should I just use my own name here? Or @aecepoglu?

Reviewer 2: AFAIK we keep the original author. For instance, [`react-markdown`](https://github.com/remarkjs/react-markdown#license) has basically been redone from the ground up but the original author is still in the license. I don't have a strong opinion on this but that's how it seems to be done most of the time.

[downloads]: https://www.npmjs.com/package/unified-language-server

[size-badge]: https://img.shields.io/bundlephobia/minzip/unified-language-server.svg

[size]: https://bundlephobia.com/result?p=unified-language-server

[sponsors-badge]: https://opencollective.com/unified/sponsors/badge.svg

[backers-badge]: https://opencollective.com/unified/backers/badge.svg

[collective]: https://opencollective.com/unified

[chat-badge]: https://img.shields.io/badge/chat-discussions-success.svg

[chat]: https://github.com/unifiedjs/rehype/discussions

[npm]: https://docs.npmjs.com/cli/install

[health]: https://github.com/unifiedjs/.github

[contributing]: https://github.com/unifiedjs/.github/blob/HEAD/contributing.md

[support]: https://github.com/unifiedjs/.github/blob/HEAD/support.md

[coc]: https://github.com/unifiedjs/.github/blob/HEAD/code-of-conduct.md

[language server]: https://microsoft.github.io/language-server-protocol/

[license]: LICENSE.txt

[author]: https://wooorm.com

[redot-language-server]: https://github.com/redotjs/redot-language-server

[rehype-language-server]: https://github.com/redotjs/rehype-language-server

> [remark-language-server]: https://github.com/redotjs/remark-language-server

Author: These language servers don’t actually exist yet, but I imagine they will be implemented soon after this PR is released.

Reviewer 2: Maybe add a comment that they are coming soon instead?
